starport/services/network/join.go

```diff
@@ -0,0 +1,218 @@
+package network
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	launchtypes "github.com/tendermint/spn/x/launch/types"
+	"github.com/tendermint/starport/starport/pkg/events"
+)
+
+// Join creates the RequestAddValidator message into the SPN
+func (b *Builder) Join(
+	ctx context.Context,
+	launchID uint64,
+	peer,
+	valKeyName,
+	valAddress string,
+	gentx,
+	consPubKey []byte,
+	selfDelegation sdk.Coin,
```

Comment:
IMO we should just provide the gentx to this method and do the parsing here and offloading logics from the `cmd` package

Reply:
We need to use the account method in the cmd package to get the validator account, but we can pass the gentx info object to the method to avoid overloading the method parameters

Reply:
Mmh we can use `b.builder.account.Name` for the account name to broadcast message, isn't it?

Reply:
Yes, but the `from` address and the `validatorAddress` can be different. The join command first checks if the address exists in the starport accounts to get the account, it's not the same account injected by the builder.